services/net/reporting/ReportingManager.cs

```diff
@@ -374,47 +374,51 @@ private async Task GenerateReportAsync(ReportRequestModel request, API.Areas.Ser
             }
         }
 
+        // Save the report instance.
+        // Group content by the section name.
+        var instance = new ReportInstance(
```

Comment:
Need to generate the report instance before generating the body because the template now needs the instance primary key for a link.